blockstore/splitstore/splitstore.go

```diff
@@ -1282,6 +1298,7 @@ func (s *SplitStore) checkClosing() error {
 	return nil
 }
 
+// This is very dangerious and could go horribly wrong if we have anything else with a structure that matches a block header.
```

Comment:
Yeah I hate it too.

I've been considering removing it and just live with extra garbage from network requests until https://github.com/filecoin-project/lotus/issues/6705 is implemented.

I want to resolve that before v1.11.1 so we might as well already remove this code.

Reply:
Ok, I killed it at https://github.com/filecoin-project/lotus/pull/6474/commits/4f89d260b05e7da546b82b7e0e2f42e50851c3a3

At this point, the worst that could happen is that we keep some useless blocks from forks due to some network request.
I think we can live with that until #6705 is done.

Reply:
Eh, block headers are _tiny_ anyways.

Reply:
yeah, but they point to giant trees behind state roots :)
Anyway, it's not a big deal since we have marked the chain and it will should fixed in time for release so not really a problem.